packages/slate-react/src/components/editor.js

```diff
@@ -228,17 +189,61 @@ class Editor extends React.Component {
    */
 
   get schema() {
-    return this.state.schema
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.getSchemaWithMemoization(plugins)
   }
 
   get stack() {
-    return this.state.stack
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.getStackWithMemoization(plugins)
   }
 
   get value() {
-    return this.state.value
+    return this.processValueOnChange(this.props.value, this.stack)
   }
 
+  /**
+   * Get stack by plugins, it is a memoized function and therefore do not need
+   * to re-evaluate for the same plugins
+   *
+   * @param {Array<Object>} plugins
+   * @return {Stack}
+   */
+
+  getStackWithMemoization = memoize(plugins => {
+    this.tmp.resolves++
```

Comment:
should we move `this.tmp.resolves++` into `this.resolvePlugins` just for consistency?

Reply:
Yes.  Thank you.  Fixed~